scripts/beaker_submit_gpu_tests.py

```diff
@@ -0,0 +1,102 @@
+"""
+Usage: python -m scripts.beaker_submit_gpu_tests NAME COMMIT_SHA
+"""
+
+import signal
+import sys
+
+import rich
+from rich import pretty, print, traceback
+
+rich.get_console().width = max(rich.get_console().width, 180)
+pretty.install()
+traceback.install()
+
+from beaker import (  # noqa: E402
+    EnvVar,
+    ExperimentSpec,
+    ImageSource,
+    ResultSpec,
+    TaskContext,
+    TaskResources,
+    TaskSpec,
+)
+
+from .beaker_common import (  # noqa: E402
+    BEAKER_CLOUD_CLUSTER,
+    BEAKER_IMAGE,
+    BEAKER_ON_PREM_CLUSTERS,
+    beaker,
+)
+
+
+class TermInterrupt(Exception):
+    pass
+
+
+def handle_sigterm(sig, frame):
+    raise TermInterrupt
+
+
+def main(name: str, commit_sha: str):
+    signal.signal(signal.SIGTERM, handle_sigterm)
+
+    print(f"- Authenticated as {beaker.account.name}", end="\n\n")
+
+    # Find a cluster to use. We default to using our scalable cloud cluster,
+    # but we'll also check to see if we can find an on-prem cluster with enough
+    # free GPUs.
+    beaker_cluster = BEAKER_CLOUD_CLUSTER
+    for on_prem_cluster in BEAKER_ON_PREM_CLUSTERS:
+        for node_util in beaker.cluster.utilization(on_prem_cluster):
+            if node_util.free.gpu_count is not None and node_util.free.gpu_count >= 2:
+                beaker_cluster = on_prem_cluster
+                print(
+                    f"- Found on-prem cluster with enough free GPUs ({on_prem_cluster})", end="\n\n"
+                )
```

Comment:
This ability should really be part of Beaker.

Reply:
Yeup, they're working on it.